lib/chef/provider/windows_task.rb

```diff
@@ -201,308 +224,303 @@ def action_disable
+          current_task_trigger[:start_day].to_i != new_task_trigger[:start_day]
+        else
+          false
+        end
       end
 
-      def convert_system_date_to_mm_dd_yyyy(system_date)
-        system_date_format = convert_system_date_format_to_ruby_date_format
-        unless system_date_format == "%m/%d/%Y"
-          system_date = Date.strptime(system_date, system_date_format).strftime("%m/%d/%Y")
+      def start_time_updated?(current_task_trigger, new_task_trigger)
+        if new_resource.start_time
+          return true if current_task_trigger[:start_hour].to_i != new_task_trigger[:start_hour] ||
+          current_task_trigger[:start_minute].to_i != new_task_trigger[:start_minute]
+        else
+          false
         end
-        system_date
       end
 
-      def update_task_xml(options = [])
-        # random_delay xml element is different for different frequencies
-        random_delay_xml_element = {
-          :minute => "Triggers/TimeTrigger/RandomDelay",
-          :hourly => "Triggers/TimeTrigger/RandomDelay",
-          :once => "Triggers/TimeTrigger/RandomDelay",
-          :daily => "Triggers/CalendarTrigger/RandomDelay",
-          :weekly => "Triggers/CalendarTrigger/RandomDelay",
-          :monthly => "Triggers/CalendarTrigger/RandomDelay",
-          :none => "Triggers",
-        }
-
-        xml_element_mapping = {
-          "cwd" => "Actions/Exec/WorkingDirectory",
-          "random_delay" => random_delay_xml_element[new_resource.frequency],
-          "execution_time_limit" => "Settings/ExecutionTimeLimit",
-        }
-
-        Chef::Log.debug "looking for existing tasks"
-
-        task_script = <<-EOH
-            [Console]::OutputEncoding = [Text.UTF8Encoding]::UTF8
-            schtasks /Query /TN \"#{new_resource.task_name}\" /XML
-        EOH
-        xml_cmd = powershell_out(task_script)
-
-        return if xml_cmd.exitstatus != 0
+      def find_task
+        ts = TaskScheduler.new
+        ts.activate(new_resource.name)
+        ts.trigger(0)
+      end
 
-        doc = REXML::Document.new(xml_cmd.stdout)
+      def trigger_type
+        case new_resource.frequency
+          when :once, :minute, :hourly
+            TaskScheduler::ONCE
+          when :daily
+            TaskScheduler::DAILY
+          when :weekly
+            TaskScheduler::WEEKLY
+          when :monthly
+            # If frequency modifier is set with frequency :monthly we are setting taskscheduler as monthlydow
+            # Ref https://msdn.microsoft.com/en-us/library/windows/desktop/aa382061(v=vs.85).aspx
+            if new_resource.frequency_modifier != 1
+              TaskScheduler::MONTHLYDOW
+            else
+              TaskScheduler::MONTHLYDATE
+            end
+          when :on_idle
+            TaskScheduler::ON_IDLE
+          when :onstart
+            TaskScheduler::AT_SYSTEMSTART
+          when :on_logon
+            TaskScheduler::AT_LOGON
+          else
+            #TODO
```

Comment:
raise ArgumentError with "Invalid Trigger Type" ?

Reply:
we don't need  to raise error here since we here since we are handling this in provider